src/libraries/react-calendar/providers/calendar-context/CalendarContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+} from 'react'
+import type { CalendarState, SetCalendarState } from './CalendarContext.type'
+
+const initialCalendarState: CalendarState = {
+  selectedDate: new Date(),
+  layout: 'week',
+  openDialog: false,
+  openViewDialog: false,
+  defaultEventDuration: 60,
+  modal: false,
+  eventDialogMaxWidth: 'md',
+  fullscreen: false,
+  allowFullScreen: false,
+  withCloseIcon: true,
+  title: '',
+  content: '',
+  actions: '',
+  draggingEventId: -1,
+  startDragging: false,
+  ghostProperties: { width: 0, height: 0, date: new Date() },
+}
+
+const CalendarStateContext = createContext<CalendarState>({} as CalendarState)
+
+const SetCalendarStateContext = createContext<SetCalendarState>(
+  {} as SetCalendarState
+)
+
+export const useCalendarState = () => useContext(CalendarStateContext)
+
+export const useSetCalendarState = () => useContext(SetCalendarStateContext)
+
+export function CalendarContextProvider({ children }: { children: ReactNode }) {
+  const [stateCalendar, setStateCalendar] =
+    useState<CalendarState>(initialCalendarState)
+
+  const setStateCalendarHandler = useCallback(
+    (newState: Partial<CalendarState>) => {
+      setStateCalendar((prev): CalendarState => ({ ...prev, ...newState }))
+    },
+    []
+  )
+
+  return (
+    <CalendarStateContext.Provider value={stateCalendar}>
+      <SetCalendarStateContext.Provider value={setStateCalendarHandler}>
```

Comment:
It's a very bad Idea to create to Context for set and for the state, as you lose consistensy of your application, `setStateCalendarHandler` and `stateCalendar` are directly connected to each other, so they should be in the same context, otherwise it doesn't make sense. So for this part you should return to previous version.